simplified-app-shared/src/main/java/org/nypl/simplified/app/pdf/PdfReaderActivity.kt

```diff
@@ -0,0 +1,173 @@
+import org.nypl.simplified.app.profiles.ProfileTimeOutActivity
+import org.nypl.simplified.books.api.BookID
+import org.nypl.simplified.books.book_database.api.BookDatabaseEntryFormatHandle.BookDatabaseEntryFormatHandlePDF
+import org.nypl.simplified.books.book_database.api.BookDatabaseEntryType
+import org.nypl.simplified.books.book_database.api.BookDatabaseType
+import org.nypl.simplified.profiles.api.ProfileReadableType
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.File
+import java.io.InputStream
+
+class PdfReaderActivity : ProfileTimeOutActivity(), PdfFragmentListenerType, TableOfContentsFragmentListenerType {
+
+    companion object {
+        const val TABLE_OF_CONTENTS = "table_of_contents"
+
+        private const val PARAMS_ID = "org.nypl.pdf.android.pdfreader.PdfReaderActivity.params"
+
+        /**
+         * Factory method to start a [PdfReaderActivity]
+         */
+        fun startActivity(
+                from: Activity,
+                parameters: PdfReaderParameters
+        ) {
+
+            val b = Bundle()
+            b.putSerializable(this.PARAMS_ID, parameters)
+            val i = Intent(from, PdfReaderActivity::class.java)
+            i.putExtras(b)
+            from.startActivity(i)
+        }
+    }
+
+    private val log: Logger = LoggerFactory.getLogger(PdfReaderActivity::class.java)
+
+    // vars assigned in onCreate and passed with the intent
+    private lateinit var documentTitle: String
+    private lateinit var pdfFile: File
+    private lateinit var accountId: AccountID
+    private lateinit var id: BookID
+    private lateinit var profile: ProfileReadableType
+    private lateinit var account: AccountType
+    private lateinit var books: BookDatabaseType
+    private lateinit var entry: BookDatabaseEntryType
+    private lateinit var handle: BookDatabaseEntryFormatHandlePDF
+
+    // vars for the activity to pass back to the reader or table of contents fragment
+    private var documentPageIndex: Int = 0
+    private var tableOfContentsList: ArrayList<TableOfContentsItem> = arrayListOf()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        log.debug("onCreate")
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_pdf_reader)
+
+        val intentParams = intent?.getSerializableExtra(PARAMS_ID) as PdfReaderParameters
+        this.documentTitle = intentParams.documentTile
+        this.pdfFile = intentParams.pdfFile
+        this.accountId = intentParams.accountId
+        this.id = intentParams.id
+
+        this.profile = Simplified.getServices()
+                .profilesController
+                .profileCurrent()
+
+        this.account = profile.account(accountId)
+        this.books = account.bookDatabase
+        this.entry = books.entry(id)
+        this.handle = entry.findFormatHandle(BookDatabaseEntryFormatHandlePDF::class.java)!!
+        this.documentPageIndex = handle.format.lastReadLocation!!
```

Comment:
Could do with some error checking here: The `findFormatHandle` method and `books.entry` method might return `null` or raise exceptions. It's unlikely, but this would crash the activity.